Mathlib/AlgebraicTopology/MooreComplex.lean

```diff
@@ -60,13 +60,22 @@ variable (X : SimplicialObject C)
 
 /-- The normalized Moore complex in degree `n`, as a subobject of `X n`.
 -/
-@[simp]
 def objX : ∀ n : ℕ, Subobject (X.obj (op (SimplexCategory.mk n)))
   | 0 => ⊤
   | n + 1 => Finset.univ.inf fun k : Fin (n + 1) => kernelSubobject (X.δ k.succ)
 set_option linter.uppercaseLean3 false in
 #align algebraic_topology.normalized_Moore_complex.obj_X AlgebraicTopology.NormalizedMooreComplex.objX
 
+theorem objX_zero : objX X 0 = ⊤ :=
+  rfl
+
+theorem objX_add_one (n) :
+    objX X (n + 1) = Finset.univ.inf fun k : Fin (n + 1) => kernelSubobject (X.δ k.succ) :=
+  rfl
+
+attribute [eqns objX_zero objX_add_one] objX
+attribute [simp] objX
+
```

Comment:
Why are changes like this necessary?

Reply:
Because `objX` is unfolded to the `match` expression now.